packages/cells/src/widget.ts

```diff
@@ -1445,8 +1447,8 @@ export class MarkdownCell extends AttachmentsCell {
   /**
    * Modify the cell source to include a reference to the attachment.
    */
-  protected updateCellSourceWithAttachment(attachmentName: string) {
-    const textToBeAppended = `![${attachmentName}](attachment:${attachmentName})`;
+  protected updateCellSourceWithAttachment(attachmentName: string, URI: string) {
+    const textToBeAppended = `![${attachmentName}](attachment:${URI})`;
```

Comment:
Perhaps: 
```suggestion
    const textToBeAppended = `![${attachmentName}](attachment:${URI ?? attachmentName})`;
```

Or would it work if we encoded the name here, like:

```
    const textToBeAppended = `![${attachmentName}](attachment:${encodeURI(attachmentName))})`;
```